experiments/physical_design_with_on_the_fly_gate_design/on_the_fly_sidb_circuit_design_on_defective_surface.cpp

```
@@ -160,12 +160,14 @@ int main()  // NOLINT
             write_sqd_layout(result, fmt::format("{}/{}.sqd", layouts_folder, benchmark));
 
             // check equivalence
-            const auto miter = mockturtle::miter<mockturtle::klut_network>(mapped_network, st.gate_layout.value());
-            const auto eq    = mockturtle::equivalence_checking(*miter);
+            const auto& gate_layout = st.gate_layout.value();  // NOLINT(bugprone-unchecked-optional-access)
+            const auto  miter       = mockturtle::miter<mockturtle::klut_network>(mapped_network, gate_layout);
+            const auto  eq = mockturtle::equivalence_checking(*miter);  // NOLINT(bugprone-unchecked-optional-access)
             assert(eq.has_value());
+            const auto eq_result = *eq;  // NOLINT(bugprone-unchecked-optional-access)
 
```

Cppcheck (2.21.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)